src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -193,13 +189,49 @@ object LoadVCF {
         s"""corrupt VCF: expected final header line of format `#CHROM\tPOS\tID...'
            |  found: @1""".stripMargin, headerLine)
 
-    val sampleIds: Array[String] =
-      if (dropSamples)
-        Array.empty
-      else
-        headerLine
-          .split("\t")
-          .drop(9)
+    val sampleIds: Array[String] = headerLine.split("\t").drop(9)
+
+    VCFHeaderInfo(sampleIds, infoSignature, vaSignature, gSignature, canonicalFlags, headerLines)
+  }
+
+  def apply(hc: HailContext,
+    reader: HtsjdkRecordReader,
+    file1: String,
+    files: Array[String],
+    nPartitions: Option[Int] = None,
+    dropSamples: Boolean = false): VariantSampleMatrix[Locus, Variant, Annotation] = {
+    val sc = hc.sc
+
+    val vcfHeaders: Array[VCFHeaderInfo] = files.map(parseHeader(hc, reader, _))
+    val header1 = vcfHeaders.head
+
+    files.zip(vcfHeaders).foreach { case (file, hd) =>
+      if (!(header1.sampleIds sameElements hd.sampleIds)) {
+        val firstDifference = header1.sampleIds.zip(hd.sampleIds).zipWithIndex.dropWhile { case ((s1, s2), i) => s1 == s2 }.head
```

Comment:
I think this can be cleaned up a bit. You can do both at the same time:
```scala
header1.sampleIds.iterator.zip(hd.sampleIds.iterator)
.zipWithIndex.dropWhile(...).headOption match {
    case Some(...) => fatal
    case None => 
```